pkg/iter/cache.go

```diff
@@ -35,43 +36,52 @@ func (it *cachedIterator) Reset() {
 	it.curr = -1
 }
 
-func (it *cachedIterator) Next() bool {
-	if it.base != nil {
-		ok := it.base.Next()
-		// we're done with the base iterator.
-		if !ok {
-			it.closeErr = it.base.Close()
-			it.iterErr = it.base.Error()
-			it.base = nil
-			return false
-		}
-		// we're caching entries
-		it.cache = append(it.cache, entryWithLabels{entry: it.base.Entry(), labels: it.base.Labels()})
-		it.curr++
-		return true
+func (it *cachedIterator) Base() EntryIterator {
+	return it.base
+}
+
+func (it *cachedIterator) consumeNext() bool {
```

Comment:
I wonder if we might find a better name for this; it's not super clear how this is different from `Next()` at a casual glance.

Perhaps `consumeBase`/`consumeWrapped`?
I'd personally love it if we renamed `base` and `Base()` to `wrapped` and `Wrapped()`; might be more clear?